azurelinuxagent/ga/monitor.py

```diff
@@ -287,6 +288,19 @@ def add_sysinfo(self, event):
                 event.parameters.remove(param)
         event.parameters.extend(self.sysinfo)
 
+    def update_container_id(self, event):
+        last_known_container_id = self.protocol.client.get_container_id_from_goal_state()
```

Comment:
how does this instance of protocol gets its client's goal state updated? this instance is not the same as the one in the extension handler thread.

Reply:
You are correct. I see two possible solutions for this: 1) make `HostPluginProtocol` a static class, or 2) always read the container id from the goal state on disk.

Reply:
Went for the environment variable approach to share state between threads.